projects/core/utils/mask/create-number-mask.ts

```diff
@@ -73,10 +74,16 @@ export function tuiCreateNumberMask({
         const integerCapped = integerLimit
             ? integer.slice(0, integerLimit + thousandSeparators.length)
             : integer;
-        const integerCappedClean = integerCapped
-            .replace(TUI_NON_DIGITS_REGEXP, '')
-            .replace(/^0+(?!\.|$)/, '0');
-        const withSeparator = addThousandsSeparator(integerCappedClean, thousandSymbol);
+        const integerCappedClean = integerCapped.replace(TUI_NON_DIGITS_REGEXP, '');
+        const leadingZerosMarchArray =
+            integerCappedClean.match(TUI_LEADING_ZEROES_REGEXP) || [];
+        const leadingZerosAmount =
+            leadingZerosMarchArray.length !== 0 ? leadingZerosMarchArray[0].length : 0;
+        const integerCappedZerosClean = integerCappedClean.replace(/^0+(?!\.|$)/, '');
```

Comment:
do you need a `trim()` method of the end `.replace(/^0+(?!\.|$)/, '')`?

Reply:
added